Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -84,14 +92,346 @@ namespace ScriptAutomation
+                scriptAutomationInterface->PauseAutomation();
+
+                AZ::Render::ProfilingCaptureRequestBus::Broadcast(&AZ::Render::ProfilingCaptureRequestBus::Events::CaptureCpuFrameTime, outputFilePath);
+            };
+
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
+        }
+
+        void CapturePassPipelineStatistics(const AZStd::string& outputFilePath)
+        {
+            auto operation = [outputFilePath]()
+            {
+                auto scriptAutomationInterface = ScriptAutomationInterface::Get();
+
+                scriptAutomationInterface->StartProfilingCapture();
+                scriptAutomationInterface->PauseAutomation();
+
+                AZ::Render::ProfilingCaptureRequestBus::Broadcast(&AZ::Render::ProfilingCaptureRequestBus::Events::CapturePassPipelineStatistics, outputFilePath);
+            };
+
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
+        }
+
+        void CaptureCpuProfilingStatistics(const AZStd::string& outputFilePath)
+        {
+            auto operation = [outputFilePath]()
+            {
+
+                if (auto profilerSystem = AZ::Debug::ProfilerSystemInterface::Get(); profilerSystem)
+                {
+                    auto scriptAutomationInterface = ScriptAutomationInterface::Get();
+
+                    scriptAutomationInterface->StartProfilingCapture();
+                    scriptAutomationInterface->PauseAutomation();
+                    
+                    profilerSystem->CaptureFrame(outputFilePath);
+                }
+            };
+
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
+        }
+
+        void CaptureBenchmarkMetadata(const AZStd::string& benchmarkName, const AZStd::string& outputFilePath)
+        {
+            auto operation = [benchmarkName, outputFilePath]()
+            {
+                auto scriptAutomationInterface = ScriptAutomationInterface::Get();
+
+                scriptAutomationInterface->StartProfilingCapture();
+                scriptAutomationInterface->PauseAutomation();
+
+                AZ::Render::ProfilingCaptureRequestBus::Broadcast(&AZ::Render::ProfilingCaptureRequestBus::Events::CaptureBenchmarkMetadata, benchmarkName, outputFilePath);
+            };
+
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
+        }
+
+        AZStd::string ResolvePath(const AZStd::string& path)
+        {
+            return Utils::ResolvePath(path);
+        }
+
+        AZStd::string NormalizePath(const AZStd::string& path)
+        {
+            AZStd::string normalizedPath = path;
+            AzFramework::StringFunc::Path::Normalize(normalizedPath);
+            return normalizedPath;
+        }
+
+        float DegToRad(float degrees)
+        {
+            return AZ::DegToRad(degrees);
+        }
```

Comment:
Instead of crating a wrapper function, you could just bind the function pointer directly to the BehaviorContext.

Reply:
Done